pkg/cloudprovider/vsphere/cloud.go

```diff
@@ -89,11 +92,26 @@ func (vs *VSphere) Initialize(clientBuilder cloudprovider.ControllerClientBuilde
 	} else {
 		klog.Errorf("Kubernetes Client Init Failed: %v", err)
 	}
+	if vs.isLoadBalancerSupportEnabled() {
+		klog.Info("initializing load balancer support")
+		if loadbalancer.ClusterID == "" {
+			klog.Warning("Missing cluster id, no periodical cleanup possible")
+		}
+		vs.loadbalancer.Initialize(loadbalancer.ClusterID, client, stop)
+	}
+}
+
+func (vs *VSphere) isLoadBalancerSupportEnabled() bool {
+	_, ok := os.LookupEnv("ENABLE_ALPHA_NSXT_LB")
```

Comment:
I think it would be cleaner if the gate check is where we initialize the LB instance in buildVSphereFromConfig and keep `vs.loadbalancer` as nil if the gate doesn't allow it.